### PR TITLE
Update new agents spa

### PR DESCRIPTION
**Pull request by kritika-singh3**
- the agent-env association from config repo cannot be removed
 - the unknown env cannot be associated to other agents

### CODE REVIEW

File: server/webapp/WEB-INF/rails/webpack/views/pages/new_agents/environment_dropdown_button.tsx [hunk 1]
```diff
@@ -15,16 +15,21 @@
...
 
 import {ApiResult} from "helpers/api_request_builder";
+import _ from "lodash";
 import m from "mithril";
 import Stream from "mithril/stream";
 import {Agent} from "models/new_agent/agents";
 import {DropdownAttrs, Primary} from "views/components/buttons";
 import {TriStateCheckboxField} from "views/components/forms/input_fields";
 import Style from "views/pages/new_agents/index.scss";
+import {StaticAgentsVM} from "../../../models/new_agent/agents_vm";
```
Comment (bdpiparva) [inline server/webapp/WEB-INF/rails/webpack/views/pages/new_agents/environment_dropdown_button.tsx]: `../../../models/new_agent/agents_vm` can be changed to `models/new_agent/agents_vm`

File: server/webapp/WEB-INF/rails/webpack/views/pages/new_agents/environment_dropdown_button.tsx [hunk 2]
```diff
@@ -61,4 +71,88 @@ export class EnvironmentsDropdownButton extends AbstractDropdownButton<EnvAttrs>
...
+          this.triStateCheckboxMap.set(item, new EnvironmentInfo(TristateState.on, disabled, msg));
+          break;
+        default:
+          this.triStateCheckboxMap.set(item, new EnvironmentInfo(TristateState.indeterminate, disabled, msg));
+          break;
+      }
+    });
+  }
+
+  private static isAssociatedViaConfigRepo(envName: string, agents: Agent[]): boolean {
```
Comment (bdpiparva) [inline server/webapp/WEB-INF/rails/webpack/views/pages/new_agents/environment_dropdown_button.tsx]: No need to do this over here. Can we move it to `EnvironmentInfo`?

Comment (kritika-singh3) [reply server/webapp/WEB-INF/rails/webpack/views/pages/new_agents/environment_dropdown_button.tsx]: Refactored the method in question. Moved all the logic pertaining to envs to a single loop.

**Comment by kritika-singh3** [comment]
Screenshots:
 - For agent-env association from config-rep:
Screen Shot 2019-09-24 at 2 49 04 PM

 - For agent-unknown env association 
Screen Shot 2019-09-24 at 2 49 20 PM

**Pull request merged**